pkgs/development/tools/mini-compile-commands/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, python3
+}: rec {
+
+    wrap = env:
+      let hook = "ln -s ${package}/bin/post-wrapper-hook.sh $out/nix-support/post-wrapper-hook.sh\n";
+      in env.override (old: {
+        cc = old.cc.overrideAttrs (final: previous: {
+          installPhase = previous.installPhase + hook;});
+        extraBuildInputs = ( old.extraBuildInputs or [] ) ++ [ package ];
+      });
+
+
+    package = stdenv.mkDerivation {
+      pname = "mini_compile_commands";
+      version = "0.1";
+
+      src = fetchFromGitHub {
+        owner = "danielbarter";
+        repo = "mini_compile_commands";
+        rev = "v0.1";
+        sha256 = "0yDocHY4+DsakFXDnrCa/d4ZTGxjFCW68zB8F5GsRuo=";
+      };
+
+      python = python3;
+
+      installPhase = ''
+      mkdir $out
+      mkdir $out/bin
+      export mini_compile_commands_client=$out/bin/mini_compile_commands_client.py
+      for file in $(ls $src); do
+          substituteAll $src/$file $out/bin/$file
+          chmod 555 $out/bin/$file
+      done
+      '';
+
+      meta = with lib; {
+        homepage = "https://github.com/danielbarter/mini_compile_commands";
+        description = "Generate compile_commands.json using nixpkgs infrastructure";
+        license = licenses.gpl3;
+        maintainers = with maintainers; [ danielbarter ];
+        platforms = platforms.linux;
+      };
+
+
+    };
+}
```

Comment:
```suggestion
}:

rec {
  wrap = env:
    let
      hook = "ln -s ${package}/bin/post-wrapper-hook.sh $out/nix-support/post-wrapper-hook.sh\n";
    in
      env.override (old: {
        cc = old.cc.overrideAttrs (final: previous: {
          installPhase = previous.installPhase or "" + hook;
        });
        extraBuildInputs = old.extraBuildInputs or [] ++ [ package ];
      });

      package = stdenv.mkDerivation {
        pname = "mini_compile_commands";
        version = "0.1";

        src = fetchFromGitHub {
          owner = "danielbarter";
          repo = "mini_compile_commands";
          rev = "v${version}";
          sha256 = "0yDocHY4+DsakFXDnrCa/d4ZTGxjFCW68zB8F5GsRuo=";
        };

      installPhase = ''
        mkdir -p $out/bin
        export mini_compile_commands_client=$out/bin/mini_compile_commands_client.py
        for file in $(ls $src); do
          substituteAll $src/$file $out/bin/$file
          chmod +x $out/bin/$file
        done
      '';

      meta = with lib; {
        homepage = "https://github.com/danielbarter/mini_compile_commands";
        description = "Generate compile_commands.json using nixpkgs infrastructure";
        license = licenses.gpl3;
        maintainers = with maintainers; [ danielbarter ];
        platforms = platforms.linux;
      };
    };
}
```

Reply:
no, eyeballed it. Something wrong with it?

Reply:
no not at all! Just confused about why all the punctuation etc is highlighted

Reply:
actually this did break a few things + indentation was weird. There are two attributes here: wrap and package. python needs to be set as an attribute of package so that it gets substituted correctly in `mini-compile-commands`